### PR TITLE
Properly test request for JSON mime types

### LINKED ISSUE #746: JSON body params not available in rollbar
No idea what it's related to exactly and I don't have time to debug the gem unfortunately, however I can tell you this:
* we're using ember.js on the frontend
* `ACCEPT` & `CONTENT_TYPE` headers are both set to "application/vnd.api+json"
* `params` shown in rollbar only reflect route information (controller, action, id) and are completely missing the JSON from the request body
* rollbar gem version 2.15.5

### PR DESCRIPTION
**Pull request by scudco**
* Test `Content-Type` more liberally
* Use the correct value for `Accept` within the rack request

Fixes #746

### CODE REVIEW

**Comment by scudco** [comment]
I'd be happy to write a test for this, but I couldn't immediately get the tests running. Is there a trick beyond `bundle && rake spec`?

**Comment by sdhull** [comment]
It's worth noting that problems were caused by the mimetype for [jsonapi](https://jsonapi.org/#mime-types) which was not being recognized by `json_request?` as json

**Comment by waltjones** [comment]
@scudco The root Gemfile does work, but its rails is set to 4.2.8, which doesn't work with every Ruby. The Gemfile is set up so you should be able to change to a known good rails version for your Ruby and `rm Gemfile.lock && bundle install` with no other changes.

If you're seeing a different kind of error, let me know.

**Comment by scudco** [comment]
thanks @waltjones 

I tracked it down to how the support files are loaded. 

https://github.com/rollbar/rollbar-gem/blob/master/spec/spec_helper.rb#L51

`Dir.glob/[]` doesn't guarantee the order of the array afaik. Some of the files rely on classes defined within other files. Specifically, in my case, `spec/support/deploy_api/report.rb` ends up being `require`d before `spec/support/rollbar_api.rb`, but `spec/support/deploy_api/report.rb` depends on `RollbarAPI`.

```
NameError:
  uninitialized constant RollbarAPI
# ./spec/support/deploy_api/report.rb:4:in `<module:DeployAPI>'
# ./spec/support/deploy_api/report.rb:3:in `<top (required)>'
# ./spec/spec_helper.rb:52:in `block in <top (required)>'
# ./spec/spec_helper.rb:52:in `each'
# ./spec/spec_helper.rb:52:in `<top (required)>'
# ./spec/rollbar_spec.rb:13:in `<top (required)>'
```

I can work my way around this to write tests for this PR and open another issue for this specs-specific problem.

**Comment by waltjones** [comment]
@scudco Thanks for the detail!

**Comment by scudco** [comment]
I wrote some tests @waltjones

I wasn't 100% sure where to put them. Let me know if they belong in some other part of the test suite.

File: spec/rollbar/request_data_extractor_spec.rb [hunk 1]
```diff
@@ -270,7 +282,7 @@ class ExtractorDummy
                                   :method => 'PUT')
       end
 
-      it 'extracts the correct user IP' do
+      it 'extracts the correct body' do
```
Comment (sdhull) [inline spec/rollbar/request_data_extractor_spec.rb]: nice

**Comment by waltjones** [comment]
@scudco Tests look good. Perfect spot for them, I'd say. Thank you for the PR.

**Comment by waltjones** [comment]
@scudco Released in 2.20.0

**Pull request merged**